spark/spark1-shims/src/main/scala/org/apache/zeppelin/spark/Spark1Shims.java

```diff
@@ -81,17 +82,28 @@ public String showDataFrame(Object obj, int maxResult, InterpreterContext contex
         }
       }
 
+      InterpreterResult.Type outputType = InterpreterResult.Type.TABLE;
+      String sql = context.getLocalProperties().get("sql");
+      if (StringUtils.isNotBlank(sql) && sql.trim().toLowerCase().startsWith("explain")) {
+        outputType = InterpreterResult.Type.TEXT;
+      }
+
       StringBuilder msg = new StringBuilder();
-      msg.append("\n%table ");
+      msg.append("\n%" + outputType.toString().toLowerCase() + " ");
```

Comment:
Use the StringBuilder to concatenate the string.

Reply:
I think JVM will optimize it via StringBuilder, I write in this way just for readability purpose. 
https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java

Reply:
You may be right, but I get the following suggestion in intellij when using StringBuilder and normal string concatenation.
![String_concatenation](https://user-images.githubusercontent.com/454320/125298978-f85d6780-e328-11eb-8928-5e4c929aed72.png)

Reply:
I check the bytecode, jvm indeed do optimization here.

Reply:
I thought you were also using Intellij. Don't you get the comment in your IDE?